EssayAnalyzer.Api/Services/Foundation/Users/UserService.Exception.cs

```diff
@@ -49,11 +49,27 @@ private async ValueTask<User> TryCatch(ReturnUserFunction returnUserFunction)
         }
         catch (DbUpdateException dbUpdateException)
         {
-            var failedUserStorageException = 
+            var failedUserStorageException =
                 new FailedUserStorageException(dbUpdateException);
 
             throw CreateAndLogDependencyException(failedUserStorageException);
         }
+        catch (Exception exception)
+        {
+            var failedUserServiceException = 
+                new FailedUserServiceException(exception);
+
+            throw CreateAndLogServiceException(failedUserServiceException);
+        }
+    }
+
+    private Exception CreateAndLogServiceException(Xeption exception)
+    {
+        var userServiceException = 
+            new UserServiceException(exception);
+
+        this.loggingBroker.LogError(userServiceException);
+        return userServiceException;
```

Comment:
<img width="561" alt="Screenshot 2023-12-19 at 14 30 02" src="https://github.com/azam-amonov/EssayAnalyzer/assets/84405537/0e9686bd-c3c0-4e4b-bd24-17aaf0a4d50a">
 

var userServiceException = new UserServiceException(exception);
this.loggingBroker.LogError(userServiceException);

return userServiceException;

add CODE RUB: x